doc/Makefile

```diff
@@ -144,6 +144,9 @@ doctest:
 	@echo "Testing of doctests in the sources finished, look at the " \
 	      "results in $(BUILDDIR)/doctest/output.txt."
 
+fill_doc_check:
+	grep --include '*.html' -rn _build/html/ -e '%('
+
```

Comment:
Ideally this should not return anything if all docs have been filled properly.

I think it would be a good thing to have this kind of check somewhere in CI, but not entirely sure where.


after this?

https://github.com/nilearn/nilearn/blob/ecf7686740f236274be94fe8a0757808e366debc/.github/workflows/build-docs.yml#L305C52-L305C71

Reply:
@ymzayek any gut feeling about this one?

Reply:
thinking a bit more about this, it may be better to check that none of the public function have doc strings with unreplaced `%(...)s`.

This could another application for the utilities suggested in this PR: https://github.com/nilearn/nilearn/pull/2991

Reply:
Since we are already pretty strict on the doc build and this is easy to automate I would rather put this check in the CI. Where you suggest should work just fine. And either put the code directly in the yaml file or move the shell script to `build_tools/github` and call it from there

Reply:
kind of feels that the script should stay in the doc folder as there is a usecase to run it locally.

Reply:
I don't disagree but I'm just thinking not to start adding shell scripts here and there every time something small like this comes up. An alternative to this and my suggestion is to add the code in the script as a rule in the Makefile and then run `make fill_doc_check` instead of `bash fill_doc_check.sh`. Would that work? And then you can also use it locally that way.
Just a suggestion. Otherwise LGTM!

Reply:
ah yeah the makefile was my first approach but for the life of me I could not escape a `(` in a make file (see [here](https://github.com/nilearn/nilearn/pull/3992#discussion_r1331399741)) so I fell back on a bash script.

any idea on how to just do this in a makefile?

Note: probably worth adding / transfering the comments in the makefile as to why the recipe is there.